### ISSUE TITLE
[Issue]: nvm install 12 fails

### ISSUE THREAD

**Issue by cope** [coreybutler/nvm-windows#1179]
### What happened?

C:\Windows\System32>nvm install 12
Downloading node.js version 12.22.12 (64-bit)...
Download failed. Rolling Back.
C:\Users\predrags\AppData\Roaming\nvm\v12.22.12\node64.exe
Rollback failed. remove C:\Users\predrags\AppData\Roaming\nvm\v12.22.12\node64.exe: The process cannot access the file because it is being used by another process.
Could not download node.js v12.22.12 64-bit executable.

### What did you expect to happen?

installation to work

### Version

1.1.11 or newer (Default)

### Which version of Windows?

Windows 11+

### Which locale?

Western European

### Which shell are you running NVM4W in?

Command Prompt

### User Permissions?

Administrative Privileges, Elevated

### Is Developer Mode enabled?

No (Default)

### Relevant log/console output

```Shell
C:\Windows\System32>nvm install 12
Downloading node.js version 12.22.12 (64-bit)...
Download failed. Rolling Back.
C:\Users\predrags\AppData\Roaming\nvm\v12.22.12\node64.exe
Rollback failed. remove C:\Users\predrags\AppData\Roaming\nvm\v12.22.12\node64.exe: The process cannot access the file because it is being used by another process.
Could not download node.js v12.22.12 64-bit executable.
```

### Debug Output

```Shell
C:\Windows\System32>nvm debug
Running NVM for Windows with administrator privileges.

Administrator: Command Prompt - nvm  debug
Windows Version: 10.0 (Build 22631)

NVM4W Version:      1.1.12
NVM4W Path:         C:\Users\predrags\AppData\Roaming\nvm\nvm.exe
NVM4W Settings:     C:\Users\predrags\AppData\Roaming\nvm\settings.txt
NVM_HOME:           C:\Users\predrags\AppData\Roaming\nvm
NVM_SYMLINK:        C:\Program Files\nodejs
Node Installations: C:\Users\predrags\AppData\Roaming\nvm

Total Node.js Versions: 2
Active Node.js Version: v22.11.0

IPv6 is enabled. This can slow downloads significantly.

PROBLEMS DETECTED
-----------------
The following Node installations are invalid (missing node.exe): v12.22.12 - consider reinstalling these versions
```

**Comment by joycehorn** [comment]
This just happened to me as well (using nvm 1.1.12) with multiple versions:

```
C:\Windows\System32>nvm install 14.21.3
Downloading node.js version 14.21.3 (64-bit)...
Download failed. Rolling Back.
C:\Users\hornj\AppData\Roaming\nvm\v14.21.3\node64.exe
Rollback failed. remove C:\Users\hornj\AppData\Roaming\nvm\v14.21.3\node64.exe: The process cannot access the file because it is being used by another process.
Could not download node.js v14.21.3 64-bit executable.
```

```
C:\Windows\System32>nvm install 15.14.0
Downloading node.js version 15.14.0 (64-bit)...
Download failed. Rolling Back.
C:\Users\hornj\AppData\Roaming\nvm\v15.14.0\node64.exe
Rollback failed. remove C:\Users\hornj\AppData\Roaming\nvm\v15.14.0\node64.exe: The process cannot access the file because it is being used by another process.
Could not download node.js v15.14.0 64-bit executable.
```

```
C:\Windows\System32>nvm install 13.14.0
Downloading node.js version 13.14.0 (64-bit)...
Download failed. Rolling Back.
C:\Users\hornj\AppData\Roaming\nvm\v13.14.0\node64.exe
Rollback failed. remove C:\Users\hornj\AppData\Roaming\nvm\v13.14.0\node64.exe: The process cannot access the file because it is being used by another process.
Could not download node.js v13.14.0 64-bit executable.
```

I could successfulyy install 16.20.2

**Comment by coreybutler** [comment]
@cope @joycehorn are either of you behind a proxy? That error message only displays when the remote download server responds with an unknown HTTP status code (see [here](https://github.com/coreybutler/nvm-windows/blob/6284f0ec44040764dc269144a3433664bf089646/src/web/web.go#L203)).

**Comment by joycehorn** [comment]
@coreybutler No, not behind a proxy. The interestng thing is I was able to install v16 and v10 without errors, but could not make it work with versions 13, 14 and 15.

**Comment by coreybutler** [comment]
@joycehorn did you try all of those installations at approximately the same time? While it doesn't happen often, there are rare occasions where the download server isn't actually available. I noticed that some of the old versions of Node have updated timestamps on them (and their checksums) from just a few days ago. Again, it is rare, but it is possible the remote server wasn't responding properly during a push/update.

**Comment by joycehorn** [comment]
Just tried 14.21.3 again right now, failed again. Tried 
I am on a different network right now, no proxy as well.

```
C:\Users\hornj>nvm install 14.21.3
Downloading node.js version 14.21.3 (64-bit)...
Download failed. Rolling Back.
C:\Users\hornj\AppData\Roaming\nvm\v14.21.3\node64.exe
Rollback failed. remove C:\Users\hornj\AppData\Roaming\nvm\v14.21.3\node64.exe: The process cannot access the file because it is being used by another process.
Could not download node.js v14.21.3 64-bit executable.

C:\Users\hornj>nvm install 18.20.1
Downloading node.js version 18.20.1 (64-bit)...
Extracting node and npm...
Complete
npm v10.5.0 installed successfully.


Installation complete. If you want to use this version, type

nvm use 18.20.1
```

**Comment by coreybutler** [comment]
I'll have to dig into this a little deeper. Unfortunately though, my brand new Windows machine BSOD'd and corrupted the hard drive. I'm waiting for a replacement. I can dive deeper when I have a working machine again.

**Comment by stefan-ptrvch** [comment]
I also can't install anything older than Node 16. Getting the same error as mentioned above.

**Comment by stefan-ptrvch** [comment]
@coreybutler I think I found the bug.

Basically, in [this code snippet](https://github.com/coreybutler/nvm-windows/blob/6284f0ec44040764dc269144a3433664bf089646/src/web/web.go#L222-L234), the version prefix variable gets assigned a value with a slash at the end. Then [here ](https://github.com/coreybutler/nvm-windows/blob/6284f0ec44040764dc269144a3433664bf089646/src/web/web.go#L364) the URL gets built, but notice the additional slash in `/node.exe`. The URLs end up being like this for Node 14 and Node 19 for instance:
```
 https://nodejs.org/dist/v14.21.3/win-x64//node.exe
 https://nodejs.org/dist/v19.9.0/win-x64//node.exe
```

But then further down in the same function the URL for Node 19 gets rebuilt to ` https://nodejs.org/dist/v19.9.0/node-v19.9.0-win-x64.zip` because there's [this ](https://github.com/coreybutler/nvm-windows/blob/6284f0ec44040764dc269144a3433664bf089646/src/web/web.go#L376) check.

I've just removed the slash in first `GetFullNodeUrl` and it works for me for Node versions lower than 16.9.0.

If you want, I can open a PR with the fix. Maybe #580 was happening because of the same bug?

**Comment by coreybutler** [comment]
@stefan-ptrvch just glancing at your fine sleuthing, that makes sense. I think #580 had several causes, but I wouldn't rule this out as being one cause for some people using older versions.

If you're willing to PR a patch, that would certainly help speed up the process. I have quite a few updates to get through (the next release will be bumped to 1.2.0) once my Windows computer is working again, so any help would be appreciated!

**Comment by cope** [comment]
Very nice work, guys.

**Comment by stefan-ptrvch** [comment]
> Very nice work, guys.

Hvala buraz 😁

@coreybutler I've opened the PR and I'll also link a build here with the bugfix included if someone wants to download it before v1.2.0 gets released (not sure if this is inappropriate or something, please let me know if it is)
https://github.com/stefan-ptrvch/nvm-windows/releases/tag/v1.1.12-bugfix

**Comment by coreybutler** [comment]
Thanks @stefan-ptrvch! I'm fine with having the download... encouraged actually. I want people to have a workaround until 1.2.0 can be released. This appears to be that workaround, so thank you very much.

**Comment by cope** [comment]
The exe seems to not be signed?

![image](https://github.com/user-attachments/assets/aae62674-e135-4946-a429-32c6758f3289)

**Comment by coreybutler** [comment]
@cope - correct. That's the tradeoff for the moment.

The next release (v1.2.0) will be code-signed with a new certificate and a new steward. Historically, this project was "officially" maintained by my holding firm (Ecor Ventures LLC). Earlier this year, I created a new company dedicated to support this effort (Author Software Inc) and its successor ([Runtime](https://github.com/coreybutler/nvm-windows/wiki/Runtime)). I've worked closely with Microsoft on this transition. They changed their policies/algorithms for SmartScreen back in January, but Microsoft has verified Author Software and I believe SmartScreen filters will not pop up on new releases.

**Comment by mihai-mihailescu** [comment]
I can confirm that the build @stefan-ptrvch made fixed this issue was for me.
I was having the issue with nvm install 10.13. After installing the patch build, the same command worked.

Thank you!

**Comment by stefan-ptrvch** [comment]
@mihai-mihailescu Very welcome! Glad to have helped!

**Comment by SpSirket** [comment]
I was having same issue with version 14 and @stefan-ptrvch  solution fixed my problem thx

**Comment by Leandro201927** [comment]
> > Very nice work, guys.
> 
> Hvala buraz 😁
> 
> @coreybutler I've opened the PR and I'll also link a build here with the bugfix included if someone wants to download it before v1.2.0 gets released (not sure if this is inappropriate or something, please let me know if it is).
> 
> https://github.com/stefan-ptrvch/nvm-windows/releases/tag/v1.1.12-bugfix

this patch fixed the problem on my Windows 11 x64. thanks a lot for the patch.

**Comment by alminQSD** [comment]
Hvala @stefan-ptrvch

**Comment by popopopolo** [comment]
Hello, I was also having this issue, trying to install node 4.0.0 and 0.12.1 while using nvm 1.1.12. 
I confirm @stefan-ptrvch 's fix worked like a charm! Thank you so much, have a beautiful day !

**Comment by Ugurercin** [comment]
The patch **has fixed** the problem in my case as well; it works like a charm! No issues detected so far. Thank you! @stefan-ptrvch.

**Comment by chris-grabcad** [comment]
@stefan-ptrvch Thanks so much for fixing this, we've been having all sorts of issues on our CI recently and on local dev envs. 

I can also confirm that the fix works.

I note that it identifiex itself in the installer and via `nvm version` as 1.1.11, so I assume it's just mislabeled in the build.

**Comment by stefan-ptrvch** [comment]
@chris-grabcad Thanks for pointing out the version. I've built the master branch, which is a bit newer than v1.1.12, but I think the version number hasn't been updated in the code since v1.1.11 ([see here](https://github.com/coreybutler/nvm-windows/blob/6284f0ec44040764dc269144a3433664bf089646/src/nvm.go#L36)).

**Comment by rebeccapowell** [comment]
Warning. The patch above contains Win/wacatak.h!ml virus signature.

**Comment by stefan-ptrvch** [comment]
@rebeccapowell I think that's a false positive due to the build not being signed, since I don't have a certificate. There's a discussion about it above. By the way, the PR for this was merged into `master` so you can download it and built it yourself if you want.

**Comment by edgilman** [comment]
@coreybutler, is it possible to roll out a signed release with this patch, before the next release (v1.2.0) comes out?

**Comment by AlejandroGCorzo** [comment]
Hi guys quick question how can I do @stefan-ptrvch working patch if my computer keeps deleting all its files due to the lack of a signing certificate, how did you install it so windows doesn't take it as a trojan? maybe my question is very basic but if any of you could give me a hand I would be very grateful.

**Comment by coreybutler** [comment]
@AlejandroGCorzo if possible, add an exception to your antivirus. Also be mindful that Microsoft Controlled Folder access can also delete files.

**Comment by coreybutler** [comment]
Also, since my primary focus is on Runtime (the successor to this project), I'd really love to know why so many people are using old versions of Node. Is it all about legacy app support or is there some other reason?

**Comment by stefan-ptrvch** [comment]
I have a project for a customer from 2019. that I still support but don't actively work on, and it uses a Vuejs template that's Node 14 based.

**Comment by AlejandroGCorzo** [comment]
in my case I use it for applications in my work that were created with older versions of node and I need to change node version depending on the repository I work with. Thanks for the advice in the end I had to do what you suggested to install it @coreybutler and thanks to @stefan-ptrvch for the fix

**Comment by stefan-ptrvch** [comment]
@AlejandroGCorzo Very welcome! This is actually my first open-source contribution, and it's been a very exciting couple of days. I think I might start doing it more!

**Comment by rebeccapowell** [comment]
> Also, since my primary focus is on Runtime (the successor to this project), I'd really love to know why so many people are using old versions of Node. Is it all about legacy app support or is there some other reason?

Visual Studio 2022 still installs a really old node.js version with gulp 3.0.0 for the task runner explorer, which results in a primordials error when trying to run commands due to its dependency on graceful-fs. 

https://stackoverflow.com/questions/62713963/visual-studio-task-runner-explorer-referenceerror-primordials-is-not-defined

The only viable solution seems to be to temporarily downgrade node.js when you need to.

**Comment by mejobloggs-cw** [comment]
The fix doesnt work for me unfortunatley.

However, it works fine if I download the zip file from https://nodejs.org/en/download/releases and extract it to the nvm location, then use nvm to install it.

**Comment by coreybutler** [comment]
@mejobloggs-cw are you getting the same error originally posted or something else?

**Comment by kacperzemla** [comment]
I got the same error originally posted at work ( legacy apps ) and as I could not use the fix that @stefan-ptrvch because of security  reasons I managed to download the zip file (v12.22.12) from https://nodejs.org/en/download/releases and extracted it to the nvm folder and it worked automatically.

**Comment by AlbertRG** [comment]
@kacperzemla Could you please explain me the process to do the installation from the zip file (v12.22.12)?

**Comment by DTKeysight** [comment]
Why did this suddenly start happening on older versions of NVM though?

We've been using NVM 1.1.9 for a long time, installing Node 14 without issues. 

Then our Jenkins builds randomly started failing with the "used by another process" error without us changing anything on our side.

**Comment by coreybutler** [comment]
@Jc73CCm4 I wish I knew. I still suspect something may have changed on the source servers. It could have been something like a proxy change causing download links to resolve incorrectly, or any number of other things. The only "event" I'm aware of recently was the Node Collaborators summit. Maybe something changed there. I really don't know as I could not attend this one.

To be honest, I grow rather tired of these kinds of changes because they _do_ happen, usually unannounced, and they cause a lot of problems for a lot of people. 

We are seriously considering setting up a Node.js mirror for Runtime (the successor to this project). This would provide greater control over the network stack and provide us the ability to mitigate these problems before they affect users. This would likely need to be part of a paid offering since storage/bandwidth aren't free, and there is work involved in maintaining a mirror.

Apologies if this comes across as a mini-rant. I'm writing this more so to point out I'd also like to know the root cause, but there are things we can do if the root cause cannot be identified or if it's simply due to volatility in governance.

**Comment by abhijit-23blaze** [comment]
> Also, since my primary focus is on Runtime (the successor to this project), I'd really love to know why so many people are using old versions of Node. Is it all about legacy app support or is there some other reason?

The reason I am downgrading is because of an older repo that was using version 13 and I want to make a few changes to it, and so I think old version support is crucial as a lot of code has already been written in older versions

**Comment by michaelflore** [comment]
Until there is an official fix for this, what I did in the meantime was this as a workaround for now, in case anyone is using windows

- I needed version 14.16.1 so I downloaded the x64 zip from here https://nodejs.org/dist/v14.16.1/ 
- Then moved all the files from inside the sub folder into C:\Users\[Your User Name]\AppData\Roaming\nvm\v14.16.1

Ran nvm ls and did an npm install on the node 14 project for a check and it worked!

**Comment by stefan-ptrvch** [comment]
@mejobloggs-cw If you tried installing Node v10.16.1 with the non-patched version before trying to install with the patched version, you need to remove the "partially installed" version in `C:\Users\UserName\AppData\Roaming\nvm\v10.16.1` first.

So just go to `C:\Users\UserName\AppData\Roaming\nvm` and remove the `v10.16.1` directory. Then installing with the patched version should work hopefully!

**Comment by mejobloggs-cw** [comment]
> @mejobloggs-cw If you tried installing Node v10.16.1 with the non-patched version before trying to install with the patched version, you need to remove the "partially installed" Node version in `C:\Users\UserName\AppData\Roaming\nvm\v10.16.1` first.
> 
> So just go to `C:\Users\UserName\AppData\Roaming\nvm` and remove the `v10.16.1` directory. Then installing with the patched version should work hopefully!

It was done using NVM 1.1.12.

If i cleanup the files and try again, same issue

**Comment by RachelAldava** [comment]
@stefan-ptrvch

> Hvala buraz 😁
> 
> @coreybutler I've opened the PR and I'll also link a build here with the bugfix included if someone wants to download it before v1.2.0 gets released (not sure if this is inappropriate or something, please let me know if it is).
> 
> https://github.com/stefan-ptrvch/nvm-windows/releases/tag/v1.1.12-bugfix


Hey, I just wanted to let you know that you saved the day for me and 30 fellow CS students. Thank you.

**Comment by jdmartinezm** [comment]
I solved it by doing the following process:

running the command: **nvm on**

downloa file node.exe, of the version you need in this case:

[https://nodejs.org/en/blog/release/v14.21.3](url)

Copy and paste it into the path:

C:\Users\**[your_user]**\AppData\Roaming\nvm\v14.21.3

Then run nvm install 14.21.3 and nvm use 14.21.3

**Comment by DeadDuck** [comment]
@jdmartinezm 
> Then run nvm install 14.21.3 and nvm use 14.21.3

After creating the folder, nvm identifies the version as installed. So the 'nvm install' command will indicate that the version is already installed.

**Comment by isashi** [comment]
I faced the same problem, starting from @jdmartinezm solution (not working with npm), I solved with the following steps (example with 14.21.3, but should work with any version):

1) Download the zip files from: https://nodejs.org/dist/v14.21.3/ (you can find all the versions here https://nodejs.org/dist/)
2) Extract the files in the nvm sub folder, for example `C:\Users\**[your_user]**\AppData\Roaming\nvm\v14.21.3`
3) `nvm use 14.21.3` (you can omit `nvm install 14.21.3` because it should say it's already installed)

Since the extracted folder contains all the files related to node and not only the .exe, in this way you should be able to use `npm` command

**Comment by unilin-eliasvandewalle** [comment]
> @isashi, @jdmartinezm how do you handle the npm command? These workarounds will not make it available. I think I'll just copy the npm executable to the folder.

You're probably only installing the binary. If you grab the full install it should include it

**Comment by DeadDuck** [comment]
Yes, I tried another archive from the dist folder, and it contains the full install. Thank you.

**Comment by isashi** [comment]
> @isashi, @jdmartinezm how do you handle the npm command? These workarounds will not make it available. I think I'll just copy the npm executable to the folder.

as @unilin-eliasvandewalle said if you copy all the files in the .zip folder as described in my step you should have everything. In my case I had the same files as if I have runned `nvm install 14.21.3`

**Comment by CaitlinSweeney** [comment]
I came across this issue, (I'm mac user btw, using Windows after a while now). I was having the same issue, downloaded the recent version. If I run `nvm use`, I get a .vsb Hosting script error, but it does switch the version, but something gets thrown off, I can run `node -v` or `nvp` after (I can before i run `use` the command). Maybe and env path issues? But if I open the shell or prompt as an administrator, I can use `nvm use` no issues.

**Comment by coreybutler** [comment]
@CaitlinSweeney is it a `.vsb` error or a `.vbs` error? The extension is important because there are vbs files. Additionally, `nvp` is not a command in this project. Did you mean `nvm`? It does seem like something is incorrect with your PATH environment variable.

If you can run `nvm`, I encourage you to run `nvm debug`. It should point out some of these issues if they are a problem on your system. If none of that helps, open a new issue. I'd like to keep this one focused on the issue of installing old versions of node.

**Comment by ovflowd** [comment]
> Until there is an official fix for this, what I did in the meantime was this as a workaround for now, in case anyone is using windows
> 
> * I needed version 14.16.1 so I downloaded the x64 zip from here [nodejs.org/dist/v14.16.1](https://nodejs.org/dist/v14.16.1/)
> * Then moved all the files from inside the sub folder into C:\Users[Your User Name]\AppData\Roaming\nvm\v14.16.1
> 
> Ran nvm ls and did an npm install on the node 14 project for a check and it worked!

FYI we also distribute our official Windows binaries as MSI/Windows Installer :)

**Comment by ovflowd** [comment]
> @Jc73CCm4 I wish I knew. I still suspect something may have changed on the source servers. It could have been something like a proxy change causing download links to resolve incorrectly, or any number of other things. The only "event" I'm aware of recently was the Node Collaborators summit. Maybe something changed there. I really don't know as I could not attend this one.

Chiming in from the Node-side of things here 👋 -- we indeed, during the Collaborator Summit, deployed the long-term overhaul of our infrastructure (for reliability and performance) since now the URLs are resolved by Cloudflare Workers instead of `nginx`, which must be the issue as `nginx` probably just ignored the double `//` and resolved-em, but on the new infrastructure that's wrong (it is technically wrong regardless, curious that this was a thing before, I had no idea!)

So apologies for the issues caused by our change, we simply couldn't know that "nvm for Windows" was doing things differently.

**Comment by coreybutler** [comment]
Thanks @ovflowd - honestly, and in your defense, I wasn't even aware of the malformed URL until this issue popped up. As you can see above, the issue is resolved with @stefan-ptrvch's patch, but that won't impact anyone until they upgrade (I'll have a new version out this week). Personally, I'd love it if everyone upgraded right away, but there are millions of people using this project and I know some of them can't upgrade without lengthy approval cycles... especially the state agencies and military organizations. It will be particularly difficult in US organizations with holidays approaching.

Is there any kind of temporary thing we could do on Cloudflare workers (or perhaps a cloudflare redirect rule) we could add to support this while NVM4W users transition to v1.2.0?

Also, the primary reason the MSI installers aren't used is because they only allow one version of Node at a time. It's been that way for about a decade now. Bottom line, the permissions on Windows won't allow multiple versions to be installed that way. That's why we use the binaries directly.

**Comment by ovflowd** [comment]
> Also, the primary reason the MSI installers aren't used is because they only allow one version of Node at a time.

I get it, that's why version managers exist 👀

> Is there any kind of temporary thing we could do on Cloudflare workers (or perhaps a cloudflare redirect rule) we could add to support this while NVM4W users transition to v1.2.0?

We are thinking on fixes. Are you sure that fixes will be needed? We can make a simple fix on our side but we want an ACK that your userbase needs that fix, as the behaviour is non-standard 😅

**Comment by coreybutler** [comment]
@ovflowd I saw some of the fix suggestions on Slack. Seems like there are some feasible ones. Fixes are necessary for older versions to continue working, but not for new versions. This is really just about providing a grace period while users are notified and given an opportunity to upgrade.

**Comment by coreybutler** [comment]
@cope @joycehorn @michaelflore @stefan-ptrvch @mejobloggs-cw @DeadDuck and anyone else subscribed, please try downloading again with the **unpatched** (original) version. The fine folks at nodejs.org have implemented a temporary server-side solution to account for the bug, so everything should work normally now.

@stefan-ptrvch I'm still including your patch in the next release. The fix on the Node side is just a hack to accommodate old versions. Your sleuthing/patch set this in motion. As a result, millions of people are going to avoid some tough days. Thank you.

**Comment by joycehorn** [comment]
> @cope @joycehorn @michaelflore @stefan-ptrvch @mejobloggs-cw @DeadDuck and anyone else subscribed, please try downloading/`nvm install <old_version>` again with the **unpatched** (original) version. The fine folks at nodejs.org have implemented a temporary server-side solution to account for the bug, so everything should work normally now.
> 
> @stefan-ptrvch I'm still including your patch in the next release. The fix on the Node side is a hack to accommodate the bug with old versions. Your sleuthing/patch set this in motion. As a result, millions of people are going to avoid some tough days. Thank you.

Hi @coreybutler 

Can confirm the install of both lts as well as older node versions (tested v14 and v10 ) was successul with the unpatched version!
I appreciate the work put into figuring this out!

**Comment by unilin-eliasvandewalle** [comment]
Awesome, works for me as well. I'm not allowed to run unsigned .exe's on this computer. Thx!

**Comment by stefan-ptrvch** [comment]
@coreybutler Thank you! I can also confirm that it works.

**Comment by unilin-eliasvandewalle** [comment]
@aydinburak 
Can you try completely uninstalling nvm (using the uninstall exe) and installing the latest version, then using nvm to install the desired node versions ?

**Comment by coreybutler** [comment]
@aydinburak see the [common issues](https://github.com/coreybutler/nvm-windows/wiki/Common-Issues). The most common cause of `nvm use` failures are insufficient permissions. The guide walks you through everything. If you go through the guide and still have trouble, open a new issue (seeing as this one is related to and resolved by the recent changes at nodejs.org)

**Comment by coreybutler** [comment]
Since the modifications the nodejs.org team made to accommodate NVM4W are working, I'm going to close this issue. 

I had hoped to release v1.2.0 including @stefan-ptrvch's patch over the weekend, but I only got about 90% done. If I can't release it tomorrow or Wednesday, then I'll wait until Monday (since this is a long holiday weekend in the US).

**Comment by aydinburak** [comment]
Hello again,
I uninstalled nvm using uninstall.exe and got rid of nodejs. Then I installed nvm and node 10.16.3 version first and it worked. I haven't tried installing version 20.18.0 yet because I'm due today on a project. So thank you guys for your answers! I really appreciate it @unilin-eliasvandewalle @coreybutler

**Comment by l-georg** [comment]
Can confirm, downgrading to 1.1.12 fixed the issue for us.

**Comment by vinicius-santos-xipptech** [comment]
> I was having problems installing Node.js 12.22.12. I was using NVM 1.2.2, and I fixed it by changing the NVM version to 1.1.11.

The same here. Fixed it by downgrading nvm to v1.1.12

**Comment by davidcostadev** [comment]
I'm getting this issue when I install new versions beyond v1.1.12, this error happens to me when I try to install node v6.14.2
